### DISCUSSION TITLE
CAM Frequency

### DISCUSSION THREAD

**Discussion by RuiFilipe2802** [eclipse/mosaic#317] [Q&A] [answered]
Hello, I have this event responsible for sending CAM messages every 100ms:

 public void processEvent(Event event) {
        this.sendCam();
        ((VehicleOperatingSystem)this.getOs()).getEventManager().addEvent(((VehicleOperatingSystem)this.getOs()).getSimulationTime() + 100000000L, new EventProcessor[]{this});
    }

![image](https://user-images.githubusercontent.com/56730143/236253650-406d8149-5a9a-4fd2-9135-40cf3d7d59d6.png)

By doing that I am sending 10 messages every 1 sec and my problem is that the data is only changed every 10th message. Which means that the position/speed/... values is the same in 10 messages and changes and is the same for another 10. I was wondering if it would be possible to change the frequency the CAM messages are generated/updated.  In other words, I intend to update for example the position value every 100ms and not 1s as shown below.

![image](https://user-images.githubusercontent.com/56730143/236255384-ebed239d-3ac9-4455-badc-06d2ec4b45c7.png)

**Comment by kschrab** [comment, marked as answer]
Hi, the vehicle data is updated with each simulation step of the traffic simulation. This can be configured in the `sumo/sumo_config,json` file of your scenario. If you do not have this file already, then just create it there and fill it with the following content:

```json
{
  "updateInterval": 100
}
```
